tests/integration/testlib/cmdlib.py

```diff
@@ -49,10 +49,13 @@ def exec_cmd(cmd, env=None, stdin=None):
     )
 
     out, err = p.communicate(stdin)
+    returncode = p.returncode
+    if check_returncode and returncode:
+        raise subprocess.CalledProcessError(returncode, cmd, out)
```

Comment:
This seems odd to me.
I would expect a simple change here: On failure, raise and let the caller handle it.
The caller can use a helper function that handles the exception, but having branching here does not seem right.

`CalledProcessError` is supposed to be raised by subprocess, you raising it yourself is a smell IMO.

Reply:
Yes, you are right. Fixed, thanks!